policy/release/cve.rego

```diff
@@ -141,17 +141,79 @@ deny contains result if {
 	result := lib.result_helper(rego.metadata.chain(), [])
 }
 
+# METADATA
+# title: Deprecated task name
+# description: >-
+#   The `CLAIR_SCAN_RESULT` task name has been deprecated, and has been
+#   replaced with `SCAN_OUTPUT`. If any task results with the old name are
+#   found, this rule will raise a warning.
+# custom:
+#   short_name: deprecated_cve_result_name_warning
+#   failure_msg: CVE scan uses deprecated task name
+#   solution: >-
+#     Use the newer `SCAN_OUTPUT` task name.
+#   collections:
+#   - minimal
+#   - redhat
+#   depends_on:
+#   - attestation_type.known_attestation_type
+#
+warn contains result if {
+	_vulnerabilities_deprecated
+	result := lib.result_helper(rego.metadata.chain(), [])
+}
+
+# METADATA
+# title: Deprecated task name for unpatched vulnerabilities
+# description: >-
+#   The `CLAIR_SCAN_RESULT` task name has been deprecated, and has been
+#   replaced with `SCAN_OUTPUT`. If any task results with the old name are
+#   found, this rule will raise a warning.
+# custom:
+#   short_name: deprecated_unpatched_cve_result_name_warning
+#   failure_msg: CVE scan uses deprecated task name
+#   solution: >-
+#     Use the newer `SCAN_OUTPUT` task name, including for unpached vulnerabilities.
+#   collections:
+#   - minimal
+#   - redhat
+#   depends_on:
+#   - attestation_type.known_attestation_type
+#
+warn contains result if {
+	_unpatched_vulnerabilities_deprecated
+	result := lib.result_helper(rego.metadata.chain(), [])
+}
+
 _vulnerabilities := vulnerabilities if {
 	some result in lib.results_named(_result_name)
 	vulnerabilities := result.value.vulnerabilities
+} else := vulnerabilities if {
```

Comment:
NIT: this could be simplified as:

```rego
_vulnerabilities := vulnerabilities if {
	some result in lib.results_named(_result_name)
	vulnerabilities := result.value.vulnerabilities
} else := _vulnerabilities_deprecated
```

Reply:
I've been struggling to understand Rego's conditional logic - I tried something similar to this before, but I didn't have the `:=` assignment and it didn't work. I'll try this!

Reply:
This is basically saying assign to `_vulnerabilities` whatever value `_vulnerabilities_deprecated` has in case the block above doesn't evaluate. The `:=` after `else` is a bit odd but it is required here.

Reply:
Yep, it makes sense now that I see it. Fix is pushed.